examples/react-components/src/app/page.tsx

```diff
@@ -73,29 +89,39 @@ export default function AuthPage() {
     });
   };
 
-  const toggleSocials = (key: keyof SocialConfig) => {
+  const toggleSocials = (
+    key: keyof SocialConfig | keyof SocialConfig["providers"]
```

Comment:
can we use `SocialConfig.providers` to access?

Reply:
we can't

`Cannot access 'SocialConfig.providers' because 'SocialConfig' is a type, but not a namespace. Did you mean to retrieve the type of the property 'providers' in 'SocialConfig' with 'SocialConfig["providers"]'?`

Reply:
one pattern we could use if we really want to go the extra mile to get all this stuff strictly defined:

```javascript
namespace SocialConfigNS {
  export interface SocialConfig {
    enabled: boolean;
    providers: Providers;
  }

  export interface Providers {
    google: boolean;
    apple: boolean;
    facebook: boolean;
  }
}

...

const toggleSocials = (
  key: keyof SocialConfigNS.SocialConfig | keyof SocialConfigNS.Providers
) => { };
```

don't think it's necessary, but if you want, by all means go for it 😄